File: AfoninaOlga/Task3/ThreadPool/MyTask/IMyTask.cs
namespace ThreadPool.MyTask;

public interface IMyTask<out TResult> : IDisposable
{
    bool IsCompleted { get; }

    TResult Result { get; }

    void Run();

    IMyTask<TNewResult> ContinueWith<TNewResult>(Func<TResult, TNewResult> continuation);
}

File: ErshovVladislav/Task3/src/main/java/ershov/mytask/IMyTask.java
package main.java.ershov.mytask;

import java.util.function.Function;

public interface IMyTask<TResult> extends Runnable {

    <TNewResult> IMyTask<TNewResult> continueWith(Function<TResult, TNewResult> fun);

    boolean isCompleted();

    String getFunctionErrorMessage();

    TResult result() throws InterruptedException, AggregateException;

    void setFunctionErrorMessage(String message);

}
